### PR TITLE
Updates to the Callout's documentation

### PR DESCRIPTION
**Pull request by tyrasm**
Added more robust design documentation along with a v1 do/don't table

### CODE REVIEW

**Comment by mikewheaton** [comment]
Approved 👍

File: src/documentation/pages/Callout/Callout.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
 # Callout
-Notification area that provides a detailed description or set of actions on an object.
+Callouts are a powerful way to simplify a user interface. They host tips and other information users need when they need it, with minimal effort on their part. Callouts can help you use screen space more effectively and reduce screen clutter. However, poorly designed Callouts can be annoying, distracting, unhelpful, overwhelming, or in the way. 
+
+Use a Callout for displaying additional contextual information about an item on the screen. Unlike Tooltips, Callouts also have a tail that identifies their source. A common use for Callout is the introduction of a new feature or capability of an app or site. Alternate usages include pairing the Callout with a button or clickable element for on-demand presentation of additional or supporting content.
```
Comment (ericthompson) [inline src/documentation/pages/Callout/Callout.md]: Did we want to call it a "tail" or a "beak"?

File: src/documentation/pages/Callout/Callout.md [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+
+### Best Practices
+Do | Don't
+--- | ---
+Use Callouts to introduce new concepts in an experience where highlighting specific pieces of the UI is necessary. | Don’t overuse Callout without putting the user in control. Too many Callouts which open automatically can be perceived as interrupting workflow and are a bad user experience. 
+Do be concise with the information you provide inside of a Callout. Short sentences or sentence fragments are best. | Don't use large, unformatted blocks of text in your Callout, they are difficult to read and overwhelming.
+Do be helpful with the tip text inside of your Callout. | Don't put obvious tip text, or text that simply repeats what is already on the screen in your Callout.
+Do limit the information inside of a Callout to supplemental information that users don't have to read. | Because the content inside of a Callout isn't always visible, don't put important or required information in a Callout. 
+Callouts should be placed near the object being described, usually at the pointer's tail or head if possible. | Don’t block important UI with the placement of your Callout, it is a poor user experience that will lead to frustration.
```
Comment (ericthompson) [inline src/documentation/pages/Callout/Callout.md]: Same - tail/beak

**Comment by mikewheaton** [comment]
@tyrasm: Any preference on calling it a tail or beak? We can always merge this in and change the terminology later on.

**Comment by ericthompson** [comment]
I'm good to move forward with what he wrote.

On Jul 10, 2016, at 10:44 AM, Mike Wheaton <notifications@github.com<mailto:notifications@github.com>> wrote:


@tyrasm<https://github.com/tyrasm>: Any preference on calling it a tail or beak? We can always merge this in and change the terminology later on.

-

**Pull request merged**